test/e2e/kubernetes/pod.go

```diff
@@ -154,6 +152,31 @@ func DeletePod(ctx context.Context, k8s kubernetes.Interface, name, namespace st
 	return waitForPodToBeDeleted(ctx, k8s, name, namespace)
 }
 
+// FetchLogs retrieves logs for all pods with the name prefix and returns it in a single string.
+func FetchLogs(ctx context.Context, k8s kubernetes.Interface, name, namespace string) (string, error) {
+	var pods *corev1.PodList
+
+	listOpts := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", "app.kubernetes.io/name", name),
```

Comment:
this seems super specific to your usecase
why don't you pass kubernetes.ListOptions to this method? you can set the label selector in the caller